app/src/main/kotlin/com/simplemobiletools/smsmessenger/helpers/ImageCompressor.kt

```diff
@@ -106,7 +106,11 @@ class ImageCompressor(private val context: Context) {
         private var iteration: Int = 0
 
         fun isSatisfied(imageFile: File): Boolean {
-            return imageFile.length() <= maxFileSize || iteration >= maxIteration
+            // If size requirement is not met and iteration is maxed
+            if(iteration >= maxIteration && imageFile.length() >= maxFileSize) {
+                throw Exception("Unable to compress image to targeted size")
```

Comment:
dont just throw such random exceptions, make it a boolean or something if it fails